Tests/UnitTests/IdentityStateTests.swift

```diff
@@ -575,10 +593,212 @@ class IdentityStateTests: XCTestCase {
+    func testUpdateAdvertisingIdentifierDoesNotUpdatesEmptyWithEZeros() {
+        assertUpdateAdvertisingIdentifierIsNotUpdated(persistedAdId: "", newAdId: IdentityConstants.Default.ZERO_ADVERTISING_ID, expectedAdId: nil)
+    }
+
+    /// Test ad ID is updated from old value to new value, and no consent event is dispatched
+    func testUpdateAdvertisingIdentifierUpdatesValidWithNewValidId() {
+        assertUpdateAdvertisingIdentifierIsUpdatedWithoutConsentChange(persistedAdId: "oldAdId", newAdId: "adId", expectedAdId: "adId")
+    }
+
+    /// Test ad ID is not updated when old and new values are the same
+    func testUpdateAdvertisingIdentifierDoesNotUpdatesValidWithSameValidId() {
+        assertUpdateAdvertisingIdentifierIsNotUpdated(persistedAdId: "adId", newAdId: "adId", expectedAdId: "adId")
+    }
+
+    /// Test ad ID is updated from valid value to empty string and consent false is dispatched
+    func testUpdateAdvertisingIdentifierUpdatesValidWithEmptyId() {
+        assertUpdateAdvertisingIdentifierIsUpdatedWithConsentChange(persistedAdId: "oldAdId", newAdId: "", expectedAdId: nil, expectedConsent: "n")
+    }
+
+    /// Test ad ID is updaed from valid value to empty string when all zeros is passed, and consent false is dispatched
+    func testUpdateAdvertisingIdentifierUpdatesValidWithZeros() {
+        assertUpdateAdvertisingIdentifierIsUpdatedWithConsentChange(persistedAdId: "oldAdId", newAdId: IdentityConstants.Default.ZERO_ADVERTISING_ID, expectedAdId: nil, expectedConsent: "n")
+    }
+
+    /// Test ad ID is updaed from all zeros to valid value and consent true is dispatched
+    func testUpdateAdvertisingIdentifierUpdatesZerosWithValidId() {
+        assertUpdateAdvertisingIdentifierIsUpdatedWithConsentChange(persistedAdId: IdentityConstants.Default.ZERO_ADVERTISING_ID, newAdId: "adId", expectedAdId: "adId", expectedConsent: "y")
+    }
+
+    /// Test ad ID is updated from all zeros to empty string and consent false is dispatched
+    func testUpdateAdvertisingIdentifierUpdatesZerosWithEmptyId() {
+        assertUpdateAdvertisingIdentifierIsUpdatedWithConsentChange(persistedAdId: IdentityConstants.Default.ZERO_ADVERTISING_ID, newAdId: "", expectedAdId: nil, expectedConsent: "n")
+    }
+
+    /// Test ad ID is updated from all zeros to empty string and consent false is dispatched; passing all zeros is converted to empty string
+    func testUpdateAdvertisingIdentifierUpdatesZerosWithZeros() {
+        assertUpdateAdvertisingIdentifierIsUpdatedWithConsentChange(persistedAdId: IdentityConstants.Default.ZERO_ADVERTISING_ID, newAdId: IdentityConstants.Default.ZERO_ADVERTISING_ID, expectedAdId: nil, expectedConsent: "n")
+    }
+
+    /// Test ad ID call is ignored if passing nil
+    func testUpdateAdvertisingIdentifierPassingNilIsIgnored() {
+        assertUpdateAdvertisingIdentifierIsNotUpdated(persistedAdId: "oldAdId", newAdId: nil, expectedAdId: "oldAdId")
+    }
+
```

Comment:
@emdobrin @kevinlind Could I get your advice on how to handle these test cases? My understanding was based on the spec for IDFA from the wiki, the three cases of `nil`, empty string, and all-zeros are treated the same logically: [setAdvertisingIdentifier](https://wiki.corp.adobe.com/pages/viewpage.action?spaceKey=adms&title=Advertising+ID+handling+in+Edge+Identity) "1. ad tracking disabled case when old adid is valid string, new adid is null, empty, or all-zeroes string"; should these test cases be updated to not expect a consent event to be dispatched? 

After updating the IdentityState.swift `shouldUpdateAdId(newAdID:)` logic to treat the three cases the same, these three test cases are failing